doc/user-manual/language/cubical.lagda.rst

```diff
@@ -1,106 +1,930 @@
-In a closed context, there are only two values of type ``I``: the two
-ends of the interval, ``i0`` and ``i1``::
+  JRefl : ∀ {ℓ} {A : Set ℓ} {x : A} (P : ∀ y → x ≡ y → Set ℓ)
+           (d : P x refl) → J P d refl ≡ d
+  JRefl P d = transportRefl d
 
-  a b : I
-  a = i0
-  b = i1
+Internally in Agda the ``transp`` operation computes by cases on the
+type, so for example for Σ-types it is computed elementwise. For path
+types it is however not yet possible to provide the computation rule
+as we need some way to remember the endpoints of the path after
+transporting it. Furthermore, this must work for arbitrary higher
+dimensional cubes (as we can iterate the path types). For this we
+introduce the "homogeneous composition operations" (``hcomp``) that
+generalize binary composition of paths to n-ary composition of higher
+dimensional cubes.
 
-Elements of the interval form a `De Morgan algebra <https://en.wikipedia.org/wiki/De_Morgan_algebra>`_,
-with minimum (``∧``), maximum (``∨``) and negation (``~``).
+
+Partial elements
+----------------
+
+In order to describe the homogeneous composition operations we need to
+be able to write partially specified n-dimensional cubes (i.e. cubes
+where some faces are missing). Given an element of the interval ``r :
+I`` there is a predicate ``IsOne`` which represents the constraint ``r
+= i1``. This comes with a proof that ``i1`` is in fact equal to ``i1``
+called ``1=1 : IsOne i1``. We use Greek letters like ``φ`` or ``ψ``
+when such an ``r`` should be thought of as being in the image of
+``IsOne``.
+
+Using this we introduce a type of partial elements called ``Partial φ
+A``, this is a special version of ``IsOne φ → A`` with a more
+extensional judgmental equality (two elements of ``Partial φ A`` are
+considered equal if they represent the same subcube, so the faces of
+the cubes can for example be given in different order and the two
+elements will still be considered the same). The idea is that
+``Partial φ A`` is the type of cubes in ``A`` that are only defined
+when ``IsOne φ``.  There is also a dependent version of this called
+``PartialP φ A`` which allows ``A`` to be defined only when ``IsOne
+φ``.
 
 .. code-block:: agda
 
-  module interval-example₁ (i j : I) where
-    data _≡_ (i : I) : I → Set where
-      reflI : i ≡ i
+  Partial : ∀ {ℓ} → I → Set ℓ → Setω
+
+  PartialP : ∀ {ℓ} → (φ : I) → Partial φ (Set ℓ) → Setω
+
+Partial elements are introduced using pattern matching:
+
+::
+
+  partialBool : ∀ i → Partial (i ∨ ~ i) Bool
+  partialBool i (i = i0) = true
+  partialBool i (i = i1) = false
 
-    infix 10 _≡_
+The term ``partialBool`` should be thought of a boolean with different
+values when ``(i = i0)`` and ``(i = i1)``. Terms of type ``Partial φ
+A`` can also be introduced using pattern matching lambdas
+(http://wiki.portal.chalmers.se/agda/pmwiki.php?n=ReferenceManual.PatternMatchingLambdas).
```

Comment:
The link should point to https://agda.readthedocs.io/en/latest/language/lambda-abstraction.html.

Reply:
Fixed